commit.go

```diff
@@ -128,13 +140,37 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 			}
 		}
 	}
-	src, err := b.makeImageRef(options.PreferredManifestType, options.Parent, exportBaseLayers, options.Squash, options.Compression, options.HistoryTimestamp)
+	src, err := b.makeImageRef(options.PreferredManifestType, options.Parent, exportBaseLayers, options.Squash, options.BlobDirectory, options.Compression, options.HistoryTimestamp)
 	if err != nil {
 		return imgID, nil, "", errors.Wrapf(err, "error computing layer digests and building metadata for container %q", b.ContainerID)
 	}
+	var maybeCachedSrc types.ImageReference = src
+	var maybeCachedDest types.ImageReference = dest
+	if options.BlobDirectory != "" {
+		compress := types.PreserveOriginal
+		if options.Compression != archive.Uncompressed {
+			compress = types.Compress
+		}
+		cache, err := blobcache.NewBlobCache(src, options.BlobDirectory, compress)
```

Comment:
AFAICS with the default behavior of `buildah commit`, `options.Compression == archive.Gzip`, i.e. when the destination is c/storage, this cache wrapper will, in `LayerInfosForCopy`, substitute cached compressed blobs for the uncached ones, and cause the destination to decompress them (twice! Once in `saveStream`, once in `storageImageDestination.PutBlob`), for not benefit whatsoever.

Actually, for the `makeImageRef`→c/storage common case, it seems to me that the caches should not be involved at all; either the data has already been cached, or this is eagerly making extra cached copies that might not ever be used.